scripts/linters/pre_commit_linter.py

```diff
@@ -0,0 +1,540 @@
+        invalid_filepaths = []
+        for filename in input_filenames:
+            if os.path.isfile(filename):
+                valid_filepaths.append(filename)
+            else:
+                invalid_filepaths.append(filename)
+        if invalid_filepaths:
+            python_utils.PRINT(
+                'The following file(s) do not exist: %s\n'
+                'Exiting.' % invalid_filepaths)
+            sys.exit(1)
+        all_filepaths = valid_filepaths
+    else:
+        all_filepaths = _get_changed_filepaths()
+    all_filepaths = [
+        filename for filename in all_filepaths if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    return all_filepaths
+
+
+def read_files(file_paths):
+    """Read all files to be checked and cache them. This will spin off multiple
+    threads to increase the efficiency.
+    """
+    threads = []
+    for file_path in file_paths:
+        thread = threading.Thread(target=FILE_CACHE.read, args=(file_path,))
+        thread.start()
+        threads.append(thread)
+
+    for thread in threads:
+        thread.join()
+
+
+def categorize_files(file_paths):
+    """Categorize all the files and store them in shared variable _FILES."""
+    all_filepaths_dict = {
+        '.py': [], '.html': [], '.ts': [], '.js': [], 'other': [], '.css': []
+    }
+    for file_path in file_paths:
+        _, extension = os.path.splitext(file_path)
+        if extension in all_filepaths_dict:
+            all_filepaths_dict[extension].append(file_path)
+        else:
+            all_filepaths_dict['other'].append(file_path)
+    _FILES.update(all_filepaths_dict)
+
+
+def _print_complete_summary_of_errors(all_messages):
+    """Print complete summary of errors."""
+    error_messages = all_messages
+    if error_messages != '':
+        python_utils.PRINT('Summary of Errors:')
+        python_utils.PRINT('----------------------------------------')
+        for message in error_messages:
+            python_utils.PRINT(message)
+
+
+def main(args=None):
+    """Main method for pre commit linter script that lints Python, JavaScript,
+    HTML, and CSS files.
+    """
+    parsed_args = _PARSER.parse_args(args=args)
+    # File extension to be linted.
+    file_extension_types = _get_file_extensions(
+        parsed_args.only_check_file_extensions)
+    # Default mode is non-verbose mode, if arguments contains --verbose flag it
+    # will be made True, which will represent verbose mode.
+    verbose_mode_enabled = bool(parsed_args.verbose)
+    all_filepaths = _get_all_filepaths(parsed_args.path, parsed_args.files)
+
+    python_utils.PRINT('Starting Linter....')
+
+    if len(all_filepaths) == 0:
+        python_utils.PRINT('---------------------------')
+        python_utils.PRINT('No files to check.')
+        python_utils.PRINT('---------------------------')
+        return
+
+    read_files(all_filepaths)
+    categorize_files(all_filepaths)
+
+    max_concurrent_runs = 1
```

Comment:
Rename the variable to `third_party_max_concurrent_runs`.

Reply:
Done.